siriuspy/siriuspy/devices/intlkctrl.py

```diff
@@ -268,68 +297,68 @@ def is_hutchB_gatevalves_closed(self):
     @property
     def is_frontend_eps_ok(self):
         """."""
-        state = True
-        state &= self.is_frontend_eps_mirror_pos_ok
-        state &= self.is_frontend_eps_temperatures_ok
-        state &= self.is_frontend_eps_vacuum_ok
-        return state
+        if self.is_frontend_eps_mirror_pos_ok and \
+                self.is_frontend_eps_mirror_pos_ok and \
+                self.is_frontend_eps_temperatures_ok and \
+                self.is_frontend_eps_vacuum_ok:
+            return True
+        else:
+            return False
 
     @property
     def is_hutchA_eps_ok(self):
         """."""
-        state = True
-        state &= self.is_hutchA_eps_dvf_pos_ok
-        state &= self.is_hutchA_eps_temperatures_ok
-        state &= self.is_hutchA_eps_vacuum_ok
-        return state
+        if self.is_hutchA_eps_dvf_pos_ok and \
+                self.is_hutchA_eps_temperatures_ok and \
+                self.is_hutchA_eps_vacuum_ok:
+            return True
+        else:
+            return False
 
     @property
     def is_hutchB_eps_ok(self):
         """."""
-        state = True
-        state &= self.is_hutchB_eps_vacuum_ok
-        return state
+        return self.is_hutchB_eps_vacuum_ok
 
     @property
     def is_beamline_eps_ok(self):
         """."""
-        state = True
-        state &= self.is_frontend_eps_ok
-        state &= self.is_hutchA_eps_ok
-        state &= self.is_hutchB_eps_ok
-        return state
+        if self.is_frontend_eps_ok and \
+                self.is_hutchA_eps_ok and \
+                self.is_hutchB_eps_ok:
+            return True
+        else:
+            return False
 
     @property
     def is_beamline_opened(self):
         """Return whether BL is opened."""
-        if not self.is_hutchA_intlk_search_done:
-            return False
-        if not self.is_hutchB_intlk_search_done:
-            return False
-        if not self.is_machine_gamma_enabled:
-            return False
-        if not self.is_beamline_eps_ok:
-            return False
-        if not self.is_frontend_shutter_eps_permission_ok:
-            return False
-        if not self.is_hutchA_shutter_eps_permission_ok:
-            return False
-        if not self.is_frontend_gamma_shutter_opened or \
-                not self.is_frontend_photon_shutter_opened:
-            return False
-        if not self.is_hutchA_gamma_shutter_opened:
+        if self.is_hutchA_intlk_search_done and \
+                self.is_hutchB_intlk_search_done and \
+                self.is_machine_gamma_enabled and \
+                self.is_beamline_eps_ok and \
+                self.is_frontend_shutter_eps_permission_ok and \
+                self.is_hutchA_shutter_eps_permission_ok and \
+                self.is_frontend_gamma_shutter_opened and \
+                self.is_frontend_photon_shutter_opened and \
+                self.is_hutchA_gamma_shutter_opened:
+            return True
+        else:
             return False
```

Comment:
you could return the if condition here

Reply:
ah, ok. It is True. or False.
done. thanks!